Java/CryptoDemo/src/main/java/com/breadwallet/cryptodemo/TransferListActivity.java

```diff
@@ -158,6 +161,36 @@ public Void visit(TransferDeletedEvent event) {
         });
     }
 
+    private List<? extends Transfer> getTransfers() {
+        List<? extends Transfer> walletTransfers = wallet.getTransfers();
+        Collections.sort(walletTransfers, (o1, o2) -> {
+            Optional<TransferConfirmation> oc1 = o1.getConfirmation();
+            Optional<TransferConfirmation> oc2 = o2.getConfirmation();
+
+            if (oc1.isPresent() && oc2.isPresent()) {
+                TransferConfirmation c1 = oc1.get();
+                TransferConfirmation c2 = oc2.get();
+
+                int blockCompare = c1.getBlockNumber().compareTo(c2.getBlockNumber());
+                int dateCompare = c1.getConfirmationTime().compareTo(c2.getConfirmationTime());
+                int indexCompare = c1.getTransactionIndex().compareTo(c2.getTransactionIndex());
+                return (blockCompare != 0
+                        ? blockCompare
+                        : (dateCompare != 0
+                        ? dateCompare
+                        : indexCompare));
+
+            } else if (oc1.isPresent()) {
+                return -1;
+            } else if (oc2.isPresent()) {
+                return 1;
+            } else {
+                return o1.hashCode() - o2.hashCode();
+            }
+        });
+        return walletTransfers;
+    }
+
```

Comment:
`indexCompare` before `dateCompare`, I think.

Reply:
Hmm, as a user, I'd be more interested in having things ordered by the timestamp, within a block, than its index in the block. Or am I misunderstanding what that index corresponds to?

Reply:
If confirmed, the timestamp is the block timestamp and thus multiple transactions in one block will have the same timestamp; it is the index within the block, that distinguishes such transactions.  The timestamp is useful when the block is not confirmed (assuming the timestamp is relevant at that point - like assigned on creation).

Reply:
Ah, I see. So in that case, given that this logic is within the case where both transfers are included, comparing timestamp is unnecessary entirely. Block and then index is all that is required, correct?

Reply:
Yes.